aya/src/maps/mod.rs

```diff
@@ -563,35 +565,37 @@ impl MapData {
         let fd = bpf_get_object(&path_string).map_err(|(_, io_error)| MapError::SyscallError {
             call: "BPF_OBJ_GET",
             io_error,
-        })? as RawFd;
-
-        let info = bpf_map_get_info_by_fd(fd).map_err(|io_error| MapError::SyscallError {
-            call: "BPF_MAP_GET_INFO_BY_FD",
-            io_error,
         })?;
 
+        let info =
+            bpf_map_get_info_by_fd(fd.as_fd()).map_err(|io_error| MapError::SyscallError {
+                call: "BPF_MAP_GET_INFO_BY_FD",
+                io_error,
+            })?;
+
         Ok(MapData {
             obj: parse_map_info(info, PinningType::ByName),
-            fd: Some(fd),
+            fd: Some(fd.into_raw_fd()),
             btf_fd: None,
             pinned: true,
         })
     }
 
-    /// Loads a map from a [`RawFd`].
+    /// Loads a map from a file descriptor.
```

Comment:
```suggestion
    /// Loads a map from an `[OwnedFd]`
```

Reply:
This goes against the earlier PR feedback from @tamird . I don't have a strong preference either way but I would lean towards not having the type in the comment since that is already obvious from the signature but I am fine with whatever you all device.

Reply:
That's fine, I didn't see the earlier review comment. Feel free to leave it as is.